DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetMatrixCache.h

```diff
@@ -46,7 +46,7 @@ class MatrixCache
   /// get the size of the cache
   int getSize() const { return mCache.size(); }
   /// assign matrix to a slot
-  void setMatrix(const T& mat, int sensID)
+  void setMatrix(T const& mat, int sensID)
```

Comment:
Hi,

the first binds only to a const lvalue refenrece, the second binds to const lvalue and rvalue reference.

Cheers
David

Reply:
@shahor02 just for reference about readability/interpretation of const: http://slashslash.info/eastconst/

Reply:
thanks @mkrzewic! Looks exactly like the manifesto of those who prefer right-to-left to left-to-right, w/o the difference in the net result.

Reply:
thanks @mkrzewic! Looks exactly like the manifesto of those who prefer right-to-left to left-to-right, w/o the difference in the net result.

Reply:
it's not that crazy, since the proposal makes it actually (consistently) readable (if you remember the compiler parses from right to left):
T const&  - ref to const T
T& const  - const ref to T (redundant, refs are always const)
btw, I think the above example should mean the same in both cases: const T& and T const& are equivalent, don't know why it should bind differently...

Reply:
hm, weird, have to try it out again, but the old one didn't compile when I passed in an rvalue reference.

Reply:
ok, now I am confused. "const foo& bar" and "foo const& bar" are the same. There is only a difference between "const foo& bar" and "foo& const bar". I cannot reproduce what compilation problem I had, probably was just stupid. I reverted that change to make it consistent with the other declarations in the file.